### DISCUSSION TITLE
Error in Sudoers file resulted in circular hell!!!

### DISCUSSION THREAD

**Discussion by m-salman-afzal** [microsoft/WSL#8597] [Q&A] [answered]
I was trying to automate the shutdown of wsl using power automate in windows and one of the commands was requiring password. So, I tried to bypass the password and added 
```
salman ALL = NOPASSWD: sh
```
in file opened using **visudo**. Unbeknownst to me, I saved using _Q_ even though it told me it had errors. So, now I cannot use sudo to enter root or edit the file and using 
```
pkexec nano /etc/sudoers
```
returns
```
Error getting authority: Error initializing authority: Could not connect: No such file or directory
```
So, I looked into it and one of the discussions told me that _dBus_ was not running and I cannot run dBus without sudo. 
So, please help this noob here or else I have to reset the distro and start all over again.

**Comment by Biswa96** [comment, marked as answer]
It is unclear what happened to that sudoers file. You can run that distribution as root user with this command `wsl.exe -u root -d <distro_name>`. Then modify or create the /etc/sudoers file. Here is the default permission

```
$ stat /etc/sudoers | grep Uid
Access: (0440/-r--r-----)  Uid: (    0/    root)   Gid: (    0/    root)
```

**Reply by m-salman-afzal** [reply]
Thanks for the solution. You saved me a lot of time.

**Reply by Biswa96** [reply]
visudo uses vi editor by default. To save and exit a file, type `:wq` and enter. It is also possible to change default text edior by using `EDITOR` environment variable. For example, `EDITOR=nano visudo` will use nano text editor.
